doc/source/cluster/kubernetes/examples/rayjob-kueue-gang-scheduling.md

```diff
@@ -0,0 +1,249 @@
+---
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: AdmissionCheck
+metadata:
+  name: rayjob-gpu
+spec:
+  controllerName: kueue.x-k8s.io/provisioning-request
+  parameters:
+    apiGroup: kueue.x-k8s.io
+    kind: ProvisioningRequestConfig
+    name: rayjob-gpu-config
+---
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: ProvisioningRequestConfig
+metadata:
+  name: rayjob-gpu-config
+spec:
+  provisioningClassName: queued-provisioning.gke.io
+  managedResources:
+  - nvidia.com/gpu
+---
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: ClusterQueue
+metadata:
+  name: "cluster-queue"
+spec:
+  namespaceSelector: {} # match all.
+  resourceGroups:
+  - coveredResources: ["cpu", "memory", "nvidia.com/gpu"]
+    flavors:
+    - name: "default-flavor"
+      resources:
+      - name: "cpu"
+        nominalQuota: 10000 # infinite quotas
+      - name: "memory"
+        nominalQuota: 10000Gi # infinite quotas
+      - name: "nvidia.com/gpu"
+        nominalQuota: 10000 # infinite quotas
+  admissionChecks:
+  - rayjob-gpu
+---
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: LocalQueue
+metadata:
+  namespace: "default"
+  name: "user-queue"
+spec:
+  clusterQueue: "cluster-queue"
+```
+
+Create the Kueue resources:
+```bash
+kubectl apply -f kueue-resources.yaml
+```
+
+:::{note}
+This example configures Kueue to orchestrate gang scheduling of GPUs. However, other resources such as CPU and memory can be used.
+:::
+
+## Deploy a RayJob
+
+Download the RayJob that executes all the steps documented in [Fine-tune a PyTorch Lightning Text Classifier](https://docs.ray.io/en/master/train/examples/lightning/lightning_cola_advanced.html). The [source code](https://github.com/ray-project/kuberay/tree/master/ray-operator/config/samples/pytorch-text-classifier) is also in the KubeRay repository.
+
+```bash
+curl -LO https://raw.githubusercontent.com/ray-project/kuberay/master/ray-operator/config/samples/pytorch-text-classifier/ray-job.pytorch-distributed-training.yaml
+```
+
+Before creating the RayJob, modify the RayJob metadata with a label
+to assign the RayJob to the LocalQueue created earlier:
+```yaml
+metadata:
+  generateName: pytorch-text-classifier-
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+```
+
+Deploy the RayJob:
+```bash
+$ kubectl create -f ray-job.pytorch-distributed-training.yaml
+rayjob.ray.io/dev-pytorch-text-classifier-r6d4p created
+```
+
+## Gang Scheduling with RayJob
+
+Here's the expected behavior when a GPU-requiring RayJob is deployed to a cluster initially lacking GPUs:
+* Kueue suspends the RayJob due to insufficient GPU resources in the cluster.
+* Kueue creates a ProvisioningRequest, specifying the GPU requirements for the RayJob.
+* The Kubernetes node autoscaler monitors ProvisioningRequests and adds nodes with GPUs as needed.
+* Once the required GPU nodes are available, the ProvisioningRequest is satisfied.
+* Kueue admits the RayJob, and its execution begins with Ray nodes scheduled on the newly provisioned nodes.
```

Comment:
```suggestion
* Kueue admits the RayJob, schedules Ray nodes on the newly provisioned nodes, and the RayJob execution begins.
```

Reply:
Does Kueue schedule the Ray nodes? If not, I need to correct this rewrite.

Reply:
Reworded to:
```
Kueue admits the RayJob, allowing Kubernetes to schedule the Ray nodes on the newly provisioned nodes, and the RayJob execution begins.
```